cuequivariance_torch/tests/primitives/segmented_polynomial_test.py

```diff
@@ -0,0 +1,404 @@
+            for inp in inputs:
+                if inp.size(0) != 1:
+                    batch_size = inp.size(0)
+                    break
+
+        if batch_size is None:
+            batch_size = 1
+
+        # print(f"batch_size: {batch_size}")
+        # create the output tensors
+        outputs = []
+        for i in range(self.polynomial.num_outputs):
+            output_dtype = None
+            if i < len(self.output_dtype_map):
+                if self.output_dtype_map[i] == -1:
+                    output_dtype = self.math_dtype
+                else:
+                    output_dtype = inputs[self.output_dtype_map[i]].dtype
+            if output_dtype is None and len(inputs) > 0:
+                output_dtype = inputs[0].dtype
+            if output_dtype is None:
+                output_dtype = self.math_dtype
+
+            output_batch_size = None
+            if i in output_shapes:
+                output_batch_size = output_shapes[i].size(0)
+            if output_batch_size is None:
+                if i in output_indices:
+                    output_batch_size = output_indices[i].size(0)
+            if output_batch_size is None:
+                output_batch_size = batch_size
+            outputs.append(
+                torch.zeros(
+                    (output_batch_size, self.polynomial.outputs[i].size),
+                    device=device,
+                    dtype=output_dtype,
+                )
+            )
+
+        inputs = [
+            input.index_select(0, input_indices[idx]) if idx in input_indices else input
+            for idx, input in enumerate(inputs)
+        ]
+
+        regular_outputs = [
+            torch.zeros(
+                (output_indices[idx].size(0), output.size(1)),
+                device=output.device,
+                dtype=output.dtype,
+            )
+            if idx in output_indices
+            else output
+            for idx, output in enumerate(outputs)
+        ]
+
+        # perform the operation
+        for op, stp in self.polynomial.operations:
+            self.perform_einsum(op, stp, inputs, regular_outputs)
+
+        for idx, (output, regular_output) in enumerate(zip(outputs, regular_outputs)):
+            if idx in output_indices:
+                output_index = (
+                    output_indices[idx]
+                    .reshape(-1, 1)
+                    .broadcast_to(regular_output.shape)
+                )
+                output.scatter_add_(0, output_index, regular_output)
+
+        return outputs
+
+    def perform_einsum(self, op, stp, inputs, outputs):
+        # select operands
+        inputs = [inputs[o] for o in op.buffers if o < self.polynomial.num_inputs]
+        outputs = [
+            (o_idx, outputs[o - self.polynomial.num_inputs])
+            for o_idx, o in enumerate(op.buffers)
+            if o >= self.polynomial.num_inputs
+        ]
+        assert len(outputs) == 1
+        o_idx, output = outputs[0]
```

Comment:
Here you can use the following code that will return you what you need
```
o_idx, o = op.output_operand_buffer(self.polynomial.num_inputs)
```

Reply:
done